document_ai_warehouse/batch-ingestion-docai-wh/load_docs.py

```diff
@@ -0,0 +1,493 @@
+    document_schemas = get_document_schemas()
+    document_schema_id = None
+
+    if not schema_name:
+        schema_name = processor.display_name
+
+    for f_uri in docai_output_list:
+        document_ai_output = docai_output_list[f_uri]
+        if f_uri in files_to_parse:
+            keys = get_key_value_pairs(document_ai_output)
+            create_new_schema = False
+
+            if schema_id:
+                document_schema_id = schema_id
+            else:
+                if schema_name in document_schemas:
+                    document_schema_id = document_schemas[schema_name]
+                    schema = dw_utils.get_document_schema(document_schema_id)
+                    if (
+                        schema
+                        and len(keys) != 0
+                        and len(schema.property_definitions) == 0
+                        and options
+                    ):
+                        create_new_schema = True
+                else:
+                    create_new_schema = True
+
+            if create_new_schema:
+                schema_path = create_mapping_schema(schema_name, keys, options)
+                new_schema_id = create_document_schema(schema_path, True)
+                if document_schema_id != new_schema_id:
+                    created_schemas.add(new_schema_id)
+                    document_schemas[schema_name] = new_schema_id
+                    document_schema_id = new_schema_id
+
+            (parent_id, reference_id) = files_to_parse[f_uri]
+            schema = dw_utils.get_document_schema(document_schema_id)
+
+            metadata_properties = get_metadata_properties(keys, schema)
+            try:
+                upload_document_gcs(
+                    f_uri,
+                    document_schema_id,
+                    parent_id,
+                    reference_id,
+                    document_ai_output,
+                    metadata_properties,
+                )
+            except Exception as ex:
+                Logger.error(f"Failed to upload {f_uri} - {ex}")
+
+    process_time = time.time() - initial_start_time
+    time_elapsed = round(process_time)
+    document_schema_str = ""
+    if len(created_schemas) > 0:
+        document_schema_str = (
+            f"  - created document schema with id {','.join(list(created_schemas))}"
+        )
+    Logger.info(
+        f"Job Completed in {str(round(time_elapsed / 60))} minute(s): \n"
+        f"{document_schema_str}  \n"
+        f"  - processed gcs files={len(processed_files)} \n"
+        f"  - created dw documents={len(document_id_list)} \n"
+        f"  - processed gcs directories={len(processed_dirs)} \n"
+        f"  - created dw directories={len(created_folders)} \n"
+    )
+
+    if len(error_files) != 0:
+        Logger.info(
+            f"Following files could not be handled (Document page number exceeding limit of 200 pages? {','.join(error_files)}"
+        )
+
```

Comment:
This main function is very crowded, I Think a lot of this could be split up into smaller functions

Reply:
Done